src/posts/lucky_0.25.cr

```diff
@@ -0,0 +1,370 @@
+
+    ```crystal
+    EmployeeQuery.new.where_boss(ManagerQuery.new)
+    ```
+
+    To go along with the `belongs_to`, we've also fixed some bugs related to using `has_many through`. This update will require a slightly different syntax.
+
+    ```crystal
+    # Before update
+    class User < BaseModel
+      table do
+        has_many posts : Post
+
+        # you specified the Symbol of the has_many method to query through
+        has_many comments : Comment, through: :posts
+      end
+    end
+
+    # After update
+    class User < BaseModel
+      table do
+        has_many posts : Post
+
+        # you specify an Array(Symbol) where the first item is the method to query through
+        # and the second item is that association's method.
+        has_many comments : Comment, through: [:posts, :comments]
+      end
+    end
+    ```
+
+    This means you can also now do a has_many though a has_many through 😬
+
+    ### Models now support SQL VIEW
+
+    Generally when we think of models, we think of database tables. The models even have a `table()` method to denote this!
+    Well now there's a `view()` method as well!
+
+    > SQL VIEWs are like tables, but generally READ-ONLY, and have no primary key. Their data is usually aggregated from other table sources.
+
+    ```crystal
+    class AdminUser < BaseModel
+      view do
+        column username : String
+        column promoted_to_admin_on : Time
+      end
+    end
+    ```
+
+    All of the columns for your `view` must be manually defined. These models do not come with any sort of `primary_key`, or `id` methods. This also means that methods like
+    `AdminUserQuery.find()`, and `admin_user.delete` won't work since these all rely on there being an `id` method. You must implement any of these methods yourself that you may
+    need.
+
+    ### Routing Changes
+
+    The LuckyRouter got a few new fancy that we're super stoked about! The first one is "glob" routing.
+
+    Glob routes are basically a route where you know the first part in the path, but the end of the route path is a variable length. For example:
+
+    ```
+    myblog.com/posts
+    myblog.com/posts/2020
+    myblog.com/posts/2020/12
+    myblog.com/posts/2020/12/25
+    ```
+
+    These routes may all just display posts, and you don't want to duplicate routes or pages for the same actions being taken. In this case, we can define a glob route
+    to catch all of these in to the same action.
+
+    ```crystal
+    class Posts::Index < BrowserAction
+      get "/posts/*:date" do
+        year, month, day = date.split('/')
+
+        html IndexPage, posts: PostQuery.new.by_date(year, month, day)
+      end
+    end
+    ```
+
+    Another great update to the router is that we can now catch duplicate routes defined. This helps you to catch mistakes in development when you're working on a larger
+    code base, and accidentally defined two conflicting routes.
+
+    ```crystal
+    # These will now raise an error
```

Comment:
```suggestion
    # These will now raise an error when starting the app
```

Reply:
These aren't compile-time, right? They're runtime errors? When I was fixing the Lucky repo, specs were running when I'd get an error.